solc/CommandLineInterface.cpp

```diff
@@ -649,32 +644,20 @@ bool CommandLineInterface::compile()
 	}
 	catch (InternalCompilerError const& _exception)
 	{
-		serr() <<
-			"Internal compiler error during compilation:" <<
-			endl <<
-			boost::diagnostic_information(_exception);
-		return false;
+		BOOST_THROW_EXCEPTION(CommandLineValidationError() << errinfo_comment("Internal compiler error during compilation:" << boost::diagnostic_information(_exception));
 	}
```

Comment:
I just noticed that you're modifying `CommandLineInterface`. We should refactor it to eventually too but we really need to have #11732 first - otherwise no one is catching these exceptions. That's why this issue is only about `CommandLineParser` for now. Please remove changes from this file and start with `CommandLineParser` instead. The only changes here should be in `CommandLineInterface::parseArguments()` - that's where the exceptions should be caught.